spec/src/main/asciidoc/providers.asciidoc

```diff
@@ -0,0 +1,52 @@
+//
+// Copyright (c) 2017 Contributors to the Eclipse Foundation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+[[programmatic_lookup]]
+== MicroProfile Rest Client Provider Registration
+
+The RestClientBuilder interface extends the `Configurable` interface from JAX-RS, allowing a user to register custom providers while its being built.  The behavior of the providers supported is defined by the JAX-RS Client API specification.  Below is a list of provider types expected to be supported by an implementation:
+
+=== ClientResponseFilter
+
+Filters of type `ClientResponseFilter` are invoked in order when a response is received from a remote service.
+
+=== ClientRequestFilter
+
+Filters of type `ClientRequestFilter` are invoked in order when a request is made to a remote service.
+
+=== MessageBodyReader
+
+The `MessageBodyReader` interface defined by JAX-RS allows the entity to be read from the API response after invocation.
+
+=== MessageBodyWriter
+
+The `MessageBodyWriter` interface defined by JAX-RS allows a request body to be written in the request for `@POST`, `@PUT` operations, as well as other HTTP methods that support bodies.
+
+=== ParamConverter
+
+The `ParamConverter` interface defined by JAX-RS allows a parameter in a resource method to be converted to a format to be used in a request or a response.
+
+=== ReaderInterceptor
+
+The `ReaderInterceptor` interface is a listener for when a read occurs against the response received from a remote service call.
+
+=== WriterInterceptor
+
+The `WriterInterceptor` interface is a listener for when a write occurs to the stream to be sent on the remote service invocation.
+
+== Provider Declaration
+
+In addition to defining providers via the client definition, interfaces may use the `@RegisterProviders` annotation to define classes to be registered as providers in addition to providers registered via the `RestClientBuilder`
```

Comment:
Can we clarify better what "in addition" is ?
Does the annotation augment what is passed in from Configurable or does it override?